src/core/thread/address_resolver.cpp

```diff
@@ -365,10 +369,28 @@ Error AddressResolver::UpdateCacheEntry(const Ip6::Address &aEid, Mac::ShortAddr
     return error;
 }
 
-void AddressResolver::AddSnoopedCacheEntry(const Ip6::Address &aEid, Mac::ShortAddress aRloc16)
+void AddressResolver::UpdateSnoopedCacheEntry(const Ip6::Address &aEid,
+                                              Mac::ShortAddress   aRloc16,
+                                              Mac::ShortAddress   aDest)
 {
-    uint16_t    numNonEvictable = 0;
-    CacheEntry *entry;
+    uint16_t          numNonEvictable = 0;
+    CacheEntry *      entry;
+    Mac::ShortAddress macAddress;
+
+    VerifyOrExit(UpdateCacheEntry(aEid, aRloc16) != kErrorNone);
+
+    VerifyOrExit(Get<Mle::MleRouter>().IsFullThreadDevice());
+
+    // Skip if the `aRloc16` (i.e., the source of the snooped message)
+    // is this device or an MTD (minimal) child of the device itself.
+
+    macAddress = Get<Mac::Mac>().GetShortAddress();
```

Comment:
```suggestion
    rloc16 = Get<Mac::Mac>().GetShortAddress();
```
nit: MAC Address usually refers to the Extended Address.

Reply:
Not necessarily. For example, we have `Mac::Address` type which can be either short or extended address. 

Also we already use `aRloc16` variable name in this method (for the cahce entry info EID-to-RLOC16) and we use this `macAddress` variable to compare against `aDest` which is defined as `Mac::ShortAddress` type.